src/utils.rs

```diff
@@ -0,0 +1,13 @@
+#[inline(always)]
+pub(crate) fn get_home_dir() -> Option<String> {
+    std::env::home_dir().and_then(|v| v.to_str().map(String::from))
```

Comment:
I really hate that every Rust project is converting warnings to errors 👎

What are warnings even good for then if CI decides that everything is buildable and testable but a newer Rust version emitted a new warning or in this case an older Rust version emitted a warning the stable version doesn't emit? 👎

In addition, most Rust projects don't want to provide a ci or contributing folder or at least a Makefile / justfile to allow contributors to test like CI does without search all steps in all workflows for yet another variation of build / check / test?

```suggestion
	#[allow(deprecated)]
    std::env::home_dir().and_then(|v| v.to_str().map(String::from))
```